### DISCUSSION TITLE
Android: Debug works, release crashes

### DISCUSSION THREAD

**Discussion by bsdRuf** [dotnet/maui#22792] [Q&A] [answered]
Hello, I converted a Xamarin project to Maui. I was debugging on physical and emulated android devices. After being done with the debugging, I wanted to release my App to the devices to test the offline mode.

Any ideas why the release acts diffrent then the debug? (Android only tested on 12.0  /12.1 / 13)

**Comment by davidortinau** [comment]
This is usually a linker related issue where the release build optimization strips out unused code to make your app faster and smaller. In Xcode > Windows > Devices and Simulators, check your device log or open the console and reproduce the crash. That should give you an indication of what is happening and then you can resolve it.

<img width="934" alt="Screenshot 2024-06-04 at 8 46 05 AM" src="https://github.com/dotnet/maui/assets/41873/2e660504-9a8c-4045-a973-00fc58d0c889">

**Reply by kj4bix** [reply]
I found that invalid resource bindings will run in debug but crash in release. I had the same issue finally checked every {StaticResouce } binding in the page and had an invalid one. Windows would throw an exception in debug, Android just ran, in release Android would crash. iOS didn't find them either, just ignored them.

**Comment by nll** [comment, marked as answer]
In Release mode the Pages / XAML files are compiled differently and some code maybe removed or transform differently. 

Check the device log, install Android Studio and look at the Logcat, maybe the crash information will help

**Reply by gabsamples6** [reply]
That is the main cause of issues debug vs release invalid resource that works in debug not in release -the problem is that if enable xamlc compiling than hot reload won’t work

**Comment by DanTravison** [comment]
Are there any tools that can be run to detect this?  I just hit this on android and ios and I had to pour through the android device log to track down the problem.   I would expect this problem to lend itself to static analysis.
